homeassistant/components/enphase_envoy/config_flow.py

```diff
@@ -156,8 +156,6 @@ async def async_step_reauth(
     ) -> ConfigFlowResult:
         """Handle configuration by re-auth."""
         self._reauth_entry = self._get_reauth_entry()
```

Comment:
We now recommend to NOT cache the reauth_entry, and request it once inside the required steps.

Looking more closely, I would actually recommend that you split out a `reauth_confirm` step from the `user` step, similar to how the `reconfigure` step is implemented.

```suggestion
```

You could also move the try...catch inside the `validate_input` routine to reduce duplicate code:
```python
envoy = validate_input(hass, host, user, pass, errors, description_placeholders)
if envoy:
    ...
    return self.async_create_entry(...)
# errors and description already filled
return self.async_show_form(step_id, schema, errors, description_placeholders)
```

Reply:
Thanks for the review. I've applied the 3 recommendations.